addons/api/addon/serializers/host-catalog.js

```diff
@@ -4,115 +4,43 @@
 
 export default class HostCatalogSerializer extends ApplicationSerializer {
   serialize(snapshot) {
-    const { compositeType, credentialType } = snapshot.record;
+    const { credentialType } = snapshot.record;
     const serialized = super.serialize(...arguments);
-
-    // This is a hack to remove the worker_filter field from the serialized data if the compositeType is not AWS
-    // Ideally we should refactor the model to include the compositeType using `for` attribute
-    if (compositeType !== TYPE_HOST_CATALOG_PLUGIN_AWS) {
-      delete serialized.worker_filter;
-    }
-
     // By default, disable credential rotation for dynamic credentials
     if (credentialType === TYPE_CREDENTIAL_DYNAMIC) {
       serialized.attributes.disable_credential_rotation = true;
     }
-
-    switch (compositeType) {
-      case 'static':
-        return this.serializeStatic(...arguments);
-      default:
-        return serialized;
-    }
+    return serialized;
   }
 
   serializeAttribute(snapshot, json, key, attribute) {
     const value = super.serializeAttribute(...arguments);
-    const { compositeType, credentialType, isPlugin } = snapshot.record;
+    const { compositeType, credentialType } = snapshot.record;
     const { options } = attribute;
-    const fields =
-      compositeType === TYPE_HOST_CATALOG_PLUGIN_AZURE
-        ? azureFields
-        : AWSfieldsWithCredentialType[credentialType];
-
-    // Delete any nested attribute fields that don't belong to the record's compositeType
-    if (isPlugin && options.isNestedAttribute && json.attributes) {
-      // If compositeType is AWS, delete azure fields from the json, and set the rest that don't belong to respective compositeType to null
-      if (compositeType === TYPE_HOST_CATALOG_PLUGIN_AWS) {
-        if (!AWSfieldsWithCredentialType[credentialType].includes(key)) {
-          if (azureFields.includes(key)) {
-            delete json.attributes[key];
-          } else {
-            json.attributes[key] = null;
-          }
-        }
-      }
-
-      // If compositeType is Azure, delete AWS fields from the json
-      if (
-        compositeType === TYPE_HOST_CATALOG_PLUGIN_AZURE &&
-        !azureFields.includes(key)
-      ) {
-        delete json.attributes[key];
-      }
-    }
-
-    // Delete any secret fields that don't belong to the record type
-    if (isPlugin && options.isNestedSecret && json.secrets) {
-      if (!fields.includes(key)) {
+    const isSamePluginName = options?.for?.name?.includes(compositeType);
+
+    // Remove attributes that are not applicable to the current credential type for the same plugin
+    // For example, 'aws' dynamic plugin has two credential types: 'static' and 'dynamic'
+    // API expects the fields to be 'null' for the fields that are not from the current credential type (that's why we are not using the application serializer for this, where we delete attrs)
+    if (
+      isSamePluginName &&
+      typeOf(options.for) === 'object' &&
```

Comment:
Isn't this implied if `isSamePluginName` returns true since it's looking for `options?.for?.name`?

Reply:
good catch, I removed the redudant check